src/frontends/common/include/openvino/frontend/graph_iterator.hpp

```diff
@@ -0,0 +1,51 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "openvino/core/any.hpp"
+#include "openvino/frontend/decoder.hpp"
+#include "openvino/frontend/visibility.hpp"
+
+namespace ov {
+namespace frontend {
+
+/// Abstract representation for an input model graph that gives nodes in topologically sorted order
+class FRONTEND_API GraphIterator : ov::RuntimeAttribute {
```

Comment:
I would suggest to hide this part for this time. It is not used in other code parts that we are really aiming to share between TF FE and TFL FE. So the whole idea behind a common tf library is to move reusable stuff _and_ APIs to support such reuse. There is no any usage of this class on both sides, so having this class common between TF FE and TFL FE artificially sticks us to one common interface but this inheritance is never used in the code, so this is a kind of false-dependency between TF FE and TFL FE.

Initially this GraphIteration was introduced as a temporary stub for something more universal. For example, ONNX, PDPD, PT FE are not using and going to use it, so it is strange to see it in common frontend headers. I think we need to make better alignment between FEs on this particular part and only after that move such stuff to common headers.

Reply:
Thank you for your valuable comment! I kept GraphIterator in tensorflow fronted. Please re-review